neuralmonkey/train.py

```diff
@@ -92,161 +32,40 @@ def _main() -> None:
-
-    cont_index = 0
-
-    while (os.path.exists(log_file)
-           or os.path.exists(ini_file)
-           or os.path.exists(git_commit_file)
-           or os.path.exists(git_diff_file)
-           or os.path.exists(variables_file_prefix)
-           or os.path.exists("{}.0".format(variables_file_prefix))):
-        cont_index += 1
-
-        args_file = "{}/args.cont-{}".format(
-            cfg.args.output, cont_index)
-        log_file = "{}/experiment.log.cont-{}".format(
-            cfg.args.output, cont_index)
-        ini_file = "{}/experiment.ini.cont-{}".format(
-            cfg.args.output, cont_index)
-        orig_ini_file = "{}/original.ini.cont-{}".format(
-            cfg.args.output, cont_index)
-        git_commit_file = "{}/git_commit.cont-{}".format(
-            cfg.args.output, cont_index)
-        git_diff_file = "{}/git_diff.cont-{}".format(
-            cfg.args.output, cont_index)
-        variables_file_prefix = "{}/variables.data.cont-{}".format(
-            cfg.args.output, cont_index)
-
-    with open(args_file, "w") as file:
-        print(" ".join(shlex.quote(a) for a in sys.argv), file=file)
-
-    cfg.save_file(ini_file)
-    copyfile(args.config, orig_ini_file)
+        exp.config.save_file(exp.get_path("experiment.ini", 0))
+        copyfile(args.config, exp.get_path("original.ini", 0))
 
-    if args.init_only:
         log("Experiment directory initialized.")
 
-        cmd = [os.path.basename(sys.argv[0]), "-f", ini_file]
+        cmd = [os.path.basename(sys.argv[0]), "-f",
+               exp.get_path("experiment.ini", 0)]
         log("To start experiment, run: {}".format(" ".join(shlex.quote(a)
                                                            for a in cmd)))
         exit(0)
 
-    Logging.set_log_file(log_file)
-
-    # this points inside the neuralmonkey/ dir inside the repo, but
-    # it does not matter for git.
-    repo_dir = os.path.dirname(os.path.realpath(__file__))
-    save_git_info(repo_dir, git_commit_file, git_diff_file)
-
-    cfg.build_model(warn_unused=True)
-
-    tf_manager = cfg.model.tf_manager
-    if cfg.model.tf_manager is None:
-        tf_manager = get_default_tf_manager()
-
-    tf_manager.init_saving(variables_file_prefix)
-
     try:
-        check_dataset_and_coders(cfg.model.train_dataset,
-                                 cfg.model.runners)
-        if isinstance(cfg.model.val_dataset, Dataset):
-            check_dataset_and_coders(cfg.model.val_dataset, cfg.model.runners)
-        else:
-            for val_dataset in cfg.model.val_dataset:
-                check_dataset_and_coders(val_dataset, cfg.model.runners)
-
-        check_unused_initializers()
-    except CheckingException as exc:
-        log(str(exc), color="red")
+        exp.train()
+    except KeyboardInterrupt:
+        raise
+    except Exception:  # pylint: disable=broad-except
+        log(traceback.format_exc(), color="red")
```

Comment:
Neni tahle řádka důvod, proč se všechny tracebacky výjimek vypisujou dvakrát? :-)

Reply:
To jsi teď zkoušel, nebo se to dělo už předtím? Důvod vypisování dvakrát bude spíš takovýhle případ, který jsem myslím někde v kódu viděl:
```python
try:
    ...
except Exception as e:
    raise MyException("Exception: " + str(e))
```

Reply:
To se dělo už předtim

Reply:
Kdykoliv cokoliv spadne, tak jde dvakrát traceback, kromě keyboard interruptu.. :-)

Reply:
Každopádně nevím, proč by za to měl moct tenhle řádek. Ten jenom způsobí, že se to vypíše do logu místo do stderr.

Reply:
No každopádně se to tady vypisuje a někde předtim taky. Asi je špatně to místo předtim. Teď bych tady s tim nic nedělal.